packages/gateway/src/gateway-provider.ts

```diff
@@ -142,8 +175,17 @@ export function createGatewayProvider(
           metadataCache = metadata;
           return metadata;
         })
-        .catch((error: unknown) => {
-          throw asGatewayError(error);
+        .catch(async (error: unknown) => {
+          try {
+            const headers = await getHeaders();
+            const authMethod =
+              'x-ai-gateway-auth-method' in headers
+                ? (headers['x-ai-gateway-auth-method'] as 'api-key' | 'oidc')
```

Comment:
Same comment re: avoid hard cast, maybe we need a helper fn to do this via zod validation since we do it in a few places